### PR TITLE
Added FullScreen functionality

### PR DESCRIPTION
**Pull request by burdittw**
Added the necessary JavaScript to allow for full screen mode to be used. This code uses the MDN Fullscreen API and also takes the F11 browser mode into account to change the icon to extend and compress depending if in full screen mode or not. I have tested this code on Edge, Firefox, and Brave with no issues.

### CODE REVIEW

File: src/ts/fullscreen.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+} from './util/index'
+
+/**
+ * Constants
+ * ====================================================
+ */
+
+const CLASS_NAME_FULLSCREEN = 'fullscreen'
+const CLASS_NAME_ICON_EXPAND = 'fa-expand-arrows-alt'
+const CLASS_NAME_ICON_COMPRESS = 'fa-compress-arrows-alt'
```
Comment (danny007in) [inline src/ts/fullscreen.ts:19]: do not use fa-
add something like data-lte-fullscrenn

Comment (burdittw) [reply src/ts/fullscreen.ts:19]: Not sure what you mean here? Those are the icon classes for the button that shows.

Comment (danny007in) [reply src/ts/fullscreen.ts:19]: > Not sure what you mean here? Those are the icon classes for the button that shows.

What if they use https://heroicons.com/

Comment (burdittw) [reply src/ts/fullscreen.ts:19]: I think I know what you are getting at. I will look at it again and update it.

**Comment by burdittw** [comment]
@danny007in

New code has been commited that allows for anything to be used as the icons as long as the fullscreen/fullscreen-icon/lte-show css classes are used. I tested with font-awesome and the heroicon svg's with no issues. Hope this works this time.

File: src/scss/_main-header.scss [hunk 2]
```diff
@@ -38,3 +38,7 @@
     // margin-top: $lte-main-header-height;
   }
 }
+
+.fullscreen-icon:not(.lte-show) {
```
Comment (REJack) [inline src/scss/_main-header.scss:42]: I would like to move this into a separate .scss file but it's not important for 3 lines.

**Review by REJack — APPROVED** [review]
The script looks nice, for me it's ok. What you think @danny007in?

File: src/scss/_main-header.scss (hunk 2 above)
Comment (REJack) [reply src/scss/_main-header.scss:42]: Alright 👍🏻

**Comment by burdittw** [comment]
@danny007in and @REJack 

Items resolved in this updated PR:

1. I have merged the new v4-dev branch into my fullscreen PR and the conflicts have been resolved.
2. Updated icons to font-awesome 6.2.1.
3. Seperated the fullscreen scss into its own file as requested by @REJack.

Ready for approval!

**Comment by burdittw** [comment]
@danny007in 

I have edited this code to match up with all your changes. The checks that are making it fail are in the base code and not my code. The lint errors are in files I have not changed so nothing I can really do about it, as I am not sure what your plan is with those items.

- I merged and fixed the conflicts with all your changes and moving dev-4 to master/main.
- I fixed lint issues with Fullscreen.ts file
- I updated the code that used to be domReady to use the new onDOMContentLoaded
- I changed out the icon links that were using font-awesome to use bootstrap icons to stay in line with the project.

Let me know if you need me to change anything on this PR. It should be ready to go and keeps with the previous approval of the fullscreen functionality.

**Comment by danny007in** [comment]
Make PR commit readable https://www.youtube.com/watch?v=gXCkYkLQ3To

**Comment by danny007in** [comment]
@burdittw thanks for your PR, we have upgraded your PR here https://github.com/ColorlibHQ/AdminLTE/blob/master/src/ts/fullscreen.ts if any bug or improvement is there please check

**Pull request closed without merging**